sdk/state/update.go

```diff
@@ -17,37 +17,29 @@ import (
 // 3. Sender calls ConfirmPayment
 // 4. Receiver calls ConfirmPayment
 
-type Payment struct {
+type CloseAgreement struct {
 	IterationNumber       int64
-	Amount                Amount
+	Balance               Amount
 	CloseSignatures       []xdr.DecoratedSignature
 	DeclarationSignatures []xdr.DecoratedSignature
-	FromInitiator         bool
-}
-
-// isEquivalent returns true if all fields for the Payments are equal not including signatures, else false.
-// Two payments that are equal may have different signatures depending on who and when this method is called.
-func (p Payment) isEquivalent(p2 Payment) bool {
-	return p.IterationNumber == p2.IterationNumber && p.Amount == p2.Amount && p.FromInitiator == p2.FromInitiator
 }
 
-func (p Payment) isEmpty() bool {
-	return p.IterationNumber == 0 && p.Amount == (Amount{}) && !p.FromInitiator && len(p.CloseSignatures) == 0 && len(p.DeclarationSignatures) == 0
+// isEquivalent returns true if all fields for the close agreements are equal not including signatures, else false.
+// Two close agreements that are equal may have different signatures depending on who and when this method is called.
+func (ca CloseAgreement) isEquivalent(ca2 CloseAgreement) bool {
+	return ca.IterationNumber == ca2.IterationNumber && ca.Balance == ca2.Balance
 }
 
-type CloseAgreement struct {
-	IterationNumber       int64
-	Balance               Amount
-	CloseSignatures       []xdr.DecoratedSignature
-	DeclarationSignatures []xdr.DecoratedSignature
+func (ca CloseAgreement) isEmpty() bool {
+	return ca.IterationNumber == 0 && ca.Balance == (Amount{}) && len(ca.CloseSignatures) == 0 && len(ca.DeclarationSignatures) == 0
 }
 
-func (c *Channel) ProposePayment(amount Amount) (Payment, error) {
+func (c *Channel) ProposePayment(amount Amount) (CloseAgreement, error) {
```

Comment:
I left the method names to be `ProposePayment / ConfirmPayment`. I think it still makes more sense than `ProposeCloseAgreement / ConfirmCloseAgreement` for the user, but open to suggestions 🤔

Reply:
I think this is the right call, at least with what we know in this moment. I'm suspicious that we might still want to return a Payment type that wraps a CloseAgreement and annotates what the CloseAgreement is, where the CloseAgreement is always determinable from the Payment annotations. We wouldn't store the Payment, we'd still only store the CloseAgreement, but we'd use the Payment in communication.

Happy to run with what you've got here for now though.